frontend/src/diagram/DiagramWebSocketContainer.tsx

```diff
@@ -617,6 +631,52 @@ export const DiagramWebSocketContainer = ({
     }
   };
 
+  const handleError = useCallback(
```

Comment:
Do you really, like really really, need `useCallback` here?

Reply:
that is because of the handleError dependence in the useEffect calls. I had this error without the useCallback:
the 'handleError' function makes the dependencies of useEffect Hook (at line 659) change on every render. To fix this, wrap the definition of 'handleError' in its own useCallback() Hook

Reply:
See a6408c7bd8b82b72a5055ae784e4ed553ab44120